lib/options.js

```diff
@@ -124,6 +124,9 @@ function assignPublishing(opts) {
 
   if (opts.ably && opts.channel) {
     try {
+      opts.ably = opts.ably === true ? {} : opts.ably || {};
+      opts.ably.log = { level: 0 };
+
```

Comment:
This looks unnecessarily complicated to me.

```javascript
opts.ably = opts.ably === true ? {} : opts.ably || {};
//                                    ^ is always truthy because of `if (opts.ably && opts.channel)`
// so
opts.ably = opts.ably === true ? {} : opts.ably;
// or
if (opts.ably === true) opts.ably = {};
opts.ably.log = { level: 0 };
```

What is the current type of `opts.ably`?
I usually avoid changing types like that because it's confusing and triggers de-optimization.
If this is to allow overriding options for Ably through `opts.ably`, how about

```javascript
const ablyOpts = {log: {level: 0}};
if (typeof opts.ably != 'boolean') Object.assign(ablyOpts, opts.ably);
```